### ISSUE TITLE
Ubuntu 20.10 Build fails due to deprecated declaration

### ISSUE THREAD

**Issue by DaveJeffery** [stardot/beebasm#49]
Ubuntu 20.10, clean install. Build fails.

kk@kk-desktop:~/.local/bin/beebasm/src$ make code
Automatically using platform linux
Compiling ... commands.cpp
commands.cpp: In member function ‘void LineParser::HandlePutFileCommon(bool)’:
commands.cpp:1757:26: error: ‘std::ios_base::streampos’ is deprecated: use 'std::streampos' instead [-Werror=deprecated-declarations]
 1757 |      ifstream::streampos p = inputFile.tellg();
      |                          ^
In file included from /usr/include/c++/10/ios:42,
                 from /usr/include/c++/10/ostream:38,
                 from /usr/include/c++/10/iostream:39,
                 from commands.cpp:25:
/usr/include/c++/10/bits/ios_base.h:481:28: note: declared here
  481 |     typedef std::streampos streampos
      |                            ^~~~~~~~~
cc1plus: all warnings being treated as errors
make[1]: *** [Makefile.inc:227: objects/commands.o] Error 1
make: *** [Makefile.inc:183: objs] Error 2

**Comment by ZornsLemma** [comment]
Thanks for reporting this, I hope the change in https://github.com/stardot/beebasm/pull/41 will fix this. I've merged that to proposed-updates.

**Comment by chriskillpack** [comment]
Does this repro on latest `proposed-updates` branch? If not then we should close this issue.

**Comment by SteveFosdick** [comment]
On the subject of compiler errors, on the proposed-updates branch, compiling on Linux (Arch) with gcc version 12.2.0, I get the following warnings:
```
In file included from lineparser.h:27,
                 from expression.cpp:33:
In static member function ‘static void StringHeader::Release(StringHeader**)’,
    inlined from ‘String::~String()’ at value.h:219:24,
    inlined from ‘void LineParser::EvalAdd()’ at expression.cpp:778:72:
value.h:60:25: error: pointer ‘header’ used after ‘void free(void*)’ [-Werror=use-after-free]
   60 |                 header->m_refCount--;
      |                 ~~~~~~~~^~~~~~~~~~
In static member function ‘static void StringHeader::Release(StringHeader**)’,
    inlined from ‘void Value::Clear()’ at value.h:399:25,
    inlined from ‘void Value::Clear()’ at value.h:395:7,
    inlined from ‘Value::~Value()’ at value.h:338:8,
    inlined from ‘void LineParser::EvalAdd()’ at expression.cpp:778:41:
value.h:63:29: note: call to ‘void free(void*)’ here
   63 |                         free(header);
      |                         ~~~~^~~~~~~~
Compiling ... main.cpp
In static member function ‘static StringHeader* StringHeader::Allocate(const char*, unsigned int)’,
    inlined from ‘static StringHeader* StringHeader::Copy(StringHeader*)’ at value.h:180:18,
    inlined from ‘static StringHeader* StringHeader::Upper(StringHeader*)’ at value.h:131:30,
    inlined from ‘String String::Upper()’ at value.h:258:46,
    inlined from ‘void LineParser::EvalUpper()’ at expression.cpp:1688:62:
value.h:43:32: error: array subscript 0 is outside array bounds of ‘StringHeader [0]’ [-Werror=array-bounds]
   43 |                 buffer[length] = 0;
      |                 ~~~~~~~~~~~~~~~^~~
```
These are both repeated for quite a few files which include the header.

**Comment by mungre** [comment]
Sorry, I've updated the tests to run on the latest Ubuntu and macOS and it's triggering a warning.   I meant to put that in a pr, had a brain fart and pushed it to proposed-updates.   I'm just pushing what I hope is a fix.

**Comment by mungre** [comment]
Both of my attempts to fix it failed so I've reverted that change for the moment.

We're currently testing on ubuntu-latest, macos-latest and ubuntu-18.04, which are really ubuntu-20.04, macos-11 and ubuntu-18.04, not the latest versions.  And 18.04 is getting deprecated.

So I changed it to ubuntu-22.04, macos-12 and ubuntu-20.04.  Unfortunately, the latest gcc in ubuntu-22.04 gives this bogus warning in the function that initialises a string.

**Comment by mungre** [comment]
I've now [done a version](https://github.com/stardot/beebasm/tree/mungre/update-test-runners3) that tests on ubuntus both older and newer than 20.10 so I think this streampos fix is good.  I'm disinclined to merge update-test-runners3 before v1.10 though.

**Comment by mungre** [comment]
P.S. [That branch](https://github.com/stardot/beebasm/tree/mungre/update-test-runners3) also fixes the new gcc warnings.

**Comment by chriskillpack** [comment]
Would you mind turning this into a PR? I'd like to add some comments but right now they are attached to individual commits and not part of "higher level work narrative" (hand waving madly here). Thanks.

**Comment by mungre** [comment]
A good point.  It's #87.

**Comment by mungre** [comment]
Fixed in v1.10.
